packages/gatsby/src/utils/develop-html-route.ts

```diff
@@ -0,0 +1,172 @@
+import report from "gatsby-cli/lib/reporter"
+import path from "path"
+import fs from "fs-extra"
+import { codeFrameColumns } from "@babel/code-frame"
+import ansiHTML from "ansi-html"
+const { trackCli } = require(`gatsby-telemetry`)
+
+import { renderHTML } from "./worker/render-html"
+import { Stage } from "../commands/types"
+
+interface IErrorPosition {
+  filename: string
+  line: number
+  row: number
+}
+
+const getPosition = function (stackObject): IErrorPosition {
+  let filename
+  let line
+  let row
+  // Because the JavaScript error stack has not yet been standardized,
+  // wrap the stack parsing in a try/catch for a soft fail if an
+  // unexpected stack is encountered.
+  try {
+    const filteredStack = stackObject.filter(function (s) {
+      return /\(.+?\)$/.test(s)
+    })
+    let splitLine
+    // For current Node & Chromium Error stacks
+    if (filteredStack.length > 0) {
+      splitLine = filteredStack[0].match(/(?:\()(.+?)(?:\))$/)[1].split(`:`)
+      // For older, future, or otherwise unexpected stacks
```

Comment:
Oh boy

Reply:
I copied this from another lib — so not really sure what this means or how relevant it is.